### PR TITLE
[WFCORE-3972] / [WFCORE-3666] / [WFCORE-3877] / [WFCORE-3819] / [WFCORE-3396] WildFly Elytron 1.5.0.Final and related feature requests.

### PR DESCRIPTION
**Pull request by darranl**
This pull request aggregates the current WildFly Elytron component upgrades and related enhancements currently in the queue and resolves conflicts encountered when merging them independently.

The changes were initially aggregated under: -
  https://github.com/wildfly/wildfly-core/pull/3384

This pull request adds the following component upgrade: -
  https://github.com/wildfly/wildfly-core/pull/3399

This pull request also adds the RFE: -
  https://issues.jboss.org/browse/WFCORE-3396
  https://developer.jboss.org/wiki/AnalysisDesign-IntegrationWithTheLetsEncryptCertificateAuthority

### CODE REVIEW

**Comment by darranl** [comment]
I have also added the fix for https://issues.jboss.org/browse/WFCORE-3974 which was the cause of the reported test failures.

**Comment by jmesnil** [comment]
@darranl I'm looking at the status of this PR and related feature request.

The current status of related issues is:

* WFCORE-3972 - Elytron 1.5.0.Final
  * [x] merged in #3399 
* WFCORE-3955 - Elytron 1.4.0.Final
  * [x] merged in #3399 
* WFCORE-3666 - Elytron alternative for RoleMappingLogingModule
  * [x] merged in #3399 
* WFCORE-3396 - Certificate authority integration
  * [x] reviewed & approved in #3392
  * [x] has all requirements
* WFCORE-3974 - AuditLogBootingSysLogTest
  * [x] reviewed & approved in #3401
* WFCORE-3877 - elytron.Configurable optional for configurable components
  * [ ] missing requirements from EAP7-1005
  * [ ] does not seem to have been properly reviewed in #3285

TL;DR

Before we can merge this PR, we need to meet requirements and review WFCORE-3877 in #3285.

@darranl Does that look correct?

**Comment by darranl** [comment]
@jmesnil strictly speaking now this is down to 2 RFEs and a test case fix - I could recreate the branch just combining those three now if you wanted.

**Comment by jmesnil** [comment]
@darranl that might be necessary as I can not rebase this PR on top of master.

Our merge workflow is:

```
$ git fetch upstream pull/3400/head:pullRequest3400
$ git co pullRequest3400
$ git rebase upstream/master
```

But the rebase failed at 2/7 `Applying: [WFCORE-3819] custom-security-event-listener into Elytron`

**Comment by darranl** [comment]
@jmesnil you should not be rebasing, you should be merging only - by rebasing you break a lot of the functionality git provides.

If you have been rebasing that may explain why I have been seeing additional merge issues that git should have dealt with.

**Comment by darranl** [comment]
@jmesnil I have combined the relevant PRs together into one: -

https://github.com/wildfly/wildfly-core/pull/3409

If we want to go with #3409 then this PR and #3384 can be closed.

Don't rebase #3409 GitHub will handle the individual PRs included in that PR without manual intervention being required.
